gitpod-io/gitpod
# cleaning up fragment usage

## Description

When using react fragments directly around text nodes: `<>some text</>` - it doesn't interact well with some browser extensions, like Google Translate, that manipulate the dom. [This article](https://javascript.plainenglish.io/react-fragments-a-potential-program-killer-91c5203f53e1) explains the problem well.

The problem reported was on the start workspace page, but I found issues anywhere a `<Button/>` transitions to a `loading` state. I've fixed how fragments are used there so there is never one around a direct text node. I've also cleaned up other places that are potential issues, or didn't need fragments at all.

## Related Issue(s)

Fixes WEB-367, WEB-442

## How to test

* Check out the video in WEB-367 for exact steps & details in reproducing, but it's essentially installing Google Translate extension. Visit start workspace page. Translate page to another language. Click start workspace after selecting a repo.
* If that works, then the problem isn't there anymore.

## Documentation


#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - brad-web-4321d055ef8</li>
	<li><b>🔗 URL</b> - <a href="https://brad-web-4321d055ef8.preview.gitpod-dev.com/workspaces" target="_blank">brad-web-4321d055ef8.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
</ul>

## Build Options

<details>
<summary>Build</summary>

      Run the build with werft instead of GHA
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

</details>

<details>
<summary>Installer</summary>

  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment</summary>

      If enabled this will build `install/preview`
      If enabled this will create the environment on GCE infra
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`
</details>

/hold
